### DISCUSSION TITLE
How to compute subgraph matches with multidimensional numeric node attributes.

### DISCUSSION THREAD

**Discussion by gouliang1997** [networkx/networkx#6260] [Q&A] [answered]
I want to get subgraph matchings for a pair of graphs with multidimensional numeric node attributes.
The method I am using is
```
nm = iso.numerical_node_match('node_feature', None)
GM = DiGraphMatcher(G1, G2 node_match=nm)
matching_list = list(GM.subgraph_isomorphisms_iter())
```
`node_feature` is fixed-length list of numbers.
But I encountered an error
```
Exception has occurred: TypeError
must be real number, not list
  File "/Mywork/make_dataset.py", line 88, in <module>
    matching_list = list(GM.subgraph_isomorphisms_iter())
```
I would like to know if method `numerical_node_match` can compute multidimensional numerical features. 

Any comments or suggestions are welcome.

**Comment by dschult** [comment, marked as answer]
A "generic_node_match" should allow you to build the right matching function.
The syntax is:  `nm = iso.generic_node_match('node_feature', None, func)`
where `func` is a function that processes the two fixed-length lists of numbers and returns True or False.

For example:
```python
import math
def func(feature1, feature2):
    return all( math.is_close(x, y) for x,y in zip(feature1, feature2))

nm = iso.generic_node_match('node_feature', None, func)
GM = DiGraphMatcher(G1, G2 node_match=nm)
matching_list = list(GM.subgraph_isomorphisms_iter())
```

**Reply by gouliang1997** [reply]
Thank you, this has solved my problem very well!
